sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -990,23 +990,30 @@ ssize_t gcoap_encode_link(const coap_resource_t *resource, char *buf,
 
 int gcoap_add_qstring(coap_pkt_t *pdu, const char *key, const char *val)
 {
+    return gcoap_add_qstring2(pdu, key, strlen(key), val, strlen(val));
```

Comment:
this is not safe and will (or can?) crash when `val = NULL` as `strlen(NULL)` is no good. Hence this should be changed to

```
    size_t val_len = (val) ? strlen(val) : 0;
    return gcoap_add_qstring2(pdu, key, strlen(key), val, val_len);
```

Reply:
also you should really check `key` here, too -- for the same reason. So add `assert(key);` at the beginning should suffice.

Reply:
Yes, the gcoap unit tests include a query with zero length value, and it segfaults as is.